networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/Eth2Network.java

```diff
@@ -63,6 +71,19 @@ private void startup() {
     blockGossipManager = new BlockGossipManager(network, eventBus, chainStorageClient);
     attestationGossipManager = new AttestationGossipManager(network, eventBus, chainStorageClient);
     aggregateGossipManager = new AggregateGossipManager(network, eventBus, chainStorageClient);
+
+    Eth2DiscoveryServiceBuilder discoveryServiceBuilder = new Eth2DiscoveryServiceBuilder();
+    eth2DiscoveryService =
+        discoveryServiceBuilder
+            .eventBus(eventBus)
+            .network(Optional.of(network))
+            .networkInterface(discoveryNetworkConfig.getNetworkInterface())
+            .port(discoveryNetworkConfig.getListenPort())
+            .privateKey(discoveryNetworkConfig.getDiscoveryPrivateKey())
+            .peers(discoveryNetworkConfig.getPeers())
+            .build();
+
+    SafeFuture.of(eth2DiscoveryService.start()).reportExceptions();
```

Comment:
Suggest also adding logic somewhere in that class that listens for newly discovery peers and asks the `network` to connect to them.

Reply:
True. added